addons/account/models/chart_template.py

```diff
@@ -986,19 +1041,64 @@ def _load_translations(self, langs=None, companies=None):
 
         translation_importer = TranslationImporter(self.env.cr, verbose=False)
+
+        # translate records that are created from the chart_template data
+
         for chart_template, chart_companies in groupby(companies, lambda c: c.chart_template):
             template_data = self.env['account.chart.template']._get_chart_template_data(chart_template)
             template_data.pop('template_data', None)
             for mname, data in template_data.items():
                 for _xml_id, record in data.items():
+                    _name_code_translation_module = record.pop('_name_code_translation_module', 'account')
                     fnames = {fname.split('@')[0] for fname in record}
                     for lang in langs:
                         for fname in fnames:
+                            if not self.env[mname]._fields[fname].translate:
+                                continue
                             value = record.get(f"{fname}@{lang}")
                             if not value:  # manage generic locale (i.e. `fr` instead of `fr_BE`)
                                 value = record.get(f"{fname}@{lang.split('_')[0]}")
+                            if not value and fname == 'name':
+                                value = code_translations.get_python_translations(_name_code_translation_module, lang).get(record[fname])
+                                if not value:  # manage generic locale (i.e. `fr` instead of `fr_BE`)
+                                    value = code_translations.get_python_translations(_name_code_translation_module, lang.split('_')[0]).get(record[fname])
                             if value:
                                 for company in chart_companies:
                                     xml_id = f"account.{company.id}_{_xml_id}"
                                     translation_importer.model_translations[mname][fname][xml_id][lang] = value
+
+        # translate the name of TEMPLATE_MODELS records that are not created from the chart_template data
+
+        # We use all modules that modify the model of self ('account.chart.template') as translation source.
+        # The MRO (w/o duplicates) of the class / model is used to order the modules.
+        # ('dict' preservers the order of insertion of the keys in Python 3.7 +)
+        files_modifying_this_model = list(dict.fromkeys([
+            getfile(cls) for cls in self.__class__.__mro__
+            if getattr(cls, '_name', None) == self.__class__._name
+        ]))
+        modules_to_use_for_code_translations = dict.fromkeys([
+            (get_resource_from_path(path) or (None, ))[0]  # module to which 'path' belongs to or 'None'
+            for path in files_modifying_this_model
+        ])
+        modules_to_use_for_code_translations.pop(None, None)
+        modules_to_use_for_code_translations = list(modules_to_use_for_code_translations)
```

Comment:
Since I do not know which module the name is defined in I just try all modules that modify the `account.chart.template`. This could lead to wrong translations or performance issues. Maybe is is too much or you have a better idea?

Reply:
I think it is too much and we should only consider `account` for what is not defined with `@template`